pkg/fpgacontroller/patcher/patcher.go

```diff
@@ -133,57 +134,17 @@ func (p *patcher) RemoveRegion(name string) {
 	delete(p.resourceModeMap, namespace+"/"+name)
 }
 
-// getRequestedResources validates the container's requirements first, then returns them as a map.
-func getRequestedResources(container corev1.Container) (map[string]int64, error) {
+func (p *patcher) validateContainer(container corev1.Container) error {
```

Comment:
This function doesn't have side effects and doesn't depend on the `p`'s state. You can make it a free function and move it to the file where it's actually used for better code locality.

Reply:
agree, I did that first but then worked around some linter errors. v2 fixes that.